src/Tokenizer/Analyzer/BlocksAnalyzer.php

```diff
@@ -0,0 +1,66 @@
+<?php
+
+/*
+ * This file is part of PHP CS Fixer.
+ *
+ * (c) Fabien Potencier <fabien@symfony.com>
+ *     Dariusz Rumiński <dariusz.ruminski@gmail.com>
+ *
+ * This source file is subject to the MIT license that is bundled
+ * with this source code in the file LICENSE.
+ */
+
+namespace PhpCsFixer\Tokenizer\Analyzer;
+
+use PhpCsFixer\Tokenizer\Token;
+use PhpCsFixer\Tokenizer\Tokens;
+
+/**
+ * @author Kuba Werłos <werlos@gmail.com>
+ *
+ * @internal
+ */
+final class BlocksAnalyzer
```

Comment:
please add some docs for it, out of the blue, i don't see it as generic analyzer that deserve an separated, reusable class

Reply:
It's more about SRP than re-usability.

Through, I was thinking about using it when working on https://github.com/FriendsOfPHP/PHP-CS-Fixer/issues/1812